openpype/hosts/blender/plugins/publish/validate_instance_empty.py

```diff
@@ -0,0 +1,27 @@
+import bpy
+
+import pyblish.api
+
+
+class ValidateInstanceEmpty(pyblish.api.InstancePlugin):
+    """Validator to verify that the instance is not empty"""
+
+    order = pyblish.api.ValidatorOrder - 0.01
+    hosts = ["blender"]
+    families = ["model", "pointcache", "rig", "camera" "layout", "blendScene"]
+    label = "Validate Instance is not Empty"
+    optional = False
+
+    def process(self, instance):
+        self.log.debug(instance)
+        self.log.debug(instance.data)
+        if instance.data["family"] == "blendScene":
+            # blendScene instances are collections
+            collection = bpy.data.collections[instance.name]
+            if not (collection.objects or collection.children):
+                raise RuntimeError(f"Instance {instance.name} is empty.")
+        else:
+            # All other instances are objects
+            asset_group = bpy.data.objects[instance.name]
+            if not asset_group.children:
+                raise RuntimeError(f"Instance {instance.name} is empty.")
```

Comment:
This could be more universal if `CollectInstances` would just store the actual instance node itself:

```python
instance.data["instance_node"] = group
```

Then here we'd just do:
```python
node = instance.data.get("instance_node")
if not node:
    return

if isinstance(node, bpy.types.Collection):
    # collection logic
elif isinstance(node, bpy.types.Object):
    # object logic
```

What do you think?

Reply:
Ah yes, that's way better! Thanks, I'll change it :)